stacker_blueprints/elasticsearch.py

```diff
@@ -0,0 +1,205 @@
+        "TrustedNetwork": {
+            "type": str,
+            "description": "CIDR block allowed to connect to the ES cluster",
+            "default": ""},
+    }
+
+    def get_allowed_actions(self):
+        return [
+            awacs.es.Action("ESHttpGet"),
+            awacs.es.Action("ESHttpHead"),
+            awacs.es.Action("ESHttpPost"),
+            awacs.es.Action("ESHttpDelete")]
+
+    def create_dns_record(self):
+        t = self.template
+        variables = self.get_variables()
+        should_create_dns = all([
+            variables["InternalZoneId"],
+            variables["InternalZoneName"],
+            variables["InternalHostName"],
+        ])
+        if should_create_dns:
+            t.add_resource(
+                route53.RecordSetType(
+                    DNS_RECORD,
+                    HostedZoneId=variables["InternalZoneId"],
+                    Comment="ES Domain CNAME Record",
+                    Name="{}.{}".format(variables["InternalHostName"],
+                                        variables["InternalZoneName"]),
+                    Type="CNAME",
+                    TTL="120",
+                    ResourceRecords=[GetAtt(ES_DOMAIN, "DomainEndpoint")],
+                ))
+            t.add_output(Output("CNAME", Value=Ref(DNS_RECORD)))
+
+    def create_domain(self):
+        t = self.template
+        variables = self.get_variables()
+        params = {
+            "ElasticsearchVersion": variables["ElasticsearchVersion"],
+        }
+
+        policy = self.create_access_policy()
+        if policy:
+            params["AccessPolicies"] = policy
+
+        # Add any optional keys to the params dict. ES didn't have great
+        # support for passing empty values for these keys when this was
+        # created.
+        optional_keys = ["AdvancedOptions", "DomainName", "EBSOptions",
+                         "SnapshotOptions", "Tags"]
+
+        for key in optional_keys:
+            optional = variables[key]
+            if optional:
+                params[key] = optional
+
+        domain = elasticsearch.Domain.from_dict(ES_DOMAIN, params)
+        t.add_resource(domain)
+        t.add_output(Output("DomainArn", Value=GetAtt(ES_DOMAIN, "DomainArn")))
+        t.add_output(Output("DomainEndpoint", Value=GetAtt(ES_DOMAIN,
+                                                           "DomainEndpoint")))
+
+    def create_roles_policy(self):
+        t = self.template
+        variables = self.get_variables()
+        statements = [
+            Statement(
+                Effect=Allow,
+                Action=self.get_allowed_actions(),
+                Resource=[Join("/", [GetAtt(ES_DOMAIN, "DomainArn"), "*"])])]
+        t.add_resource(
+            iam.PolicyType(
+                POLICY_NAME,
+                PolicyName=POLICY_NAME,
+                PolicyDocument=Policy(Statement=statements),
+                Roles=variables["Roles"]))
+
+    def create_access_policy(self):
```

Comment:
I'd change the methods that aren't called by `create_template` to have a different prefix just for clarity.  Maybe `get_access_policy` or `generate_access_policy`?